src/app/Processor/CreateProcessorPage/CreateProcessorPage.tsx

```diff
@@ -7,32 +7,114 @@ import {
 } from "@patternfly/react-core";
 import { useHistory, useParams } from "react-router-dom";
 import { useTranslation } from "react-i18next";
+import axios from "axios";
 import ProcessorEdit from "@app/Processor/ProcessorEdit/ProcessorEdit";
 import { Breadcrumb } from "@app/components/Breadcrumb/Breadcrumb";
+import { useGetBridgeApi } from "../../../hooks/useBridgesApi/useGetBridgeApi";
+import PageHeaderSkeleton from "@app/components/PageHeaderSkeleton/PageHeaderSkeleton";
+import { useAddProcessorToBridgeApi } from "../../../hooks/useProcessorsApi/useAddProcessorToBridgeApi";
+import { ProcessorRequest } from "@openapi/generated";
+import { ResponseError } from "../../../types/Error";
+import ProcessorEditSkeleton from "@app/Processor/ProcessorEdit/ProcessorEditSkeleton";
 
 const CreateProcessorPage = (): JSX.Element => {
   const { instanceId } = useParams<InstanceRouteParams>();
+  const [existingProcessorName, setExistingProcessorName] = useState<
+    string | undefined
+  >();
+  const [requestData, setRequestData] = useState<ProcessorRequest>();
   const history = useHistory();
-  const goToInstance = (): void => history.push(`/instance/${instanceId}`);
+  const goToInstance = useCallback(
+    (): void => history.push(`/instance/${instanceId}`),
+    [instanceId, history]
+  );
+  const goToHome = useCallback((): void => history.push(`/`), [history]);
   const { t } = useTranslation(["openbridgeTempDictionary"]);
 
+  const {
+    getBridge,
+    bridge,
+    isLoading: isBridgeLoading,
+    error: bridgeError,
+  } = useGetBridgeApi();
+
+  useEffect(() => {
+    getBridge(instanceId);
+  }, [getBridge, instanceId]);
+
+  useEffect(() => {
+    if (bridgeError) {
+      goToHome();
+    }
+  }, [bridgeError, goToHome]);
+
+  const {
+    addProcessorToBridge,
+    isLoading: isAddLoading,
+    processor: addedProcessor,
+    error: processorError,
+  } = useAddProcessorToBridgeApi();
+
+  const handleSave = (requestData: ProcessorRequest): void => {
+    setRequestData(requestData);
+    addProcessorToBridge(instanceId, requestData);
+  };
+
+  useEffect(() => {
+    if (addedProcessor) {
+      goToInstance();
+    }
+  }, [addedProcessor, goToInstance]);
+
+  useEffect(() => {
+    if (processorError && axios.isAxiosError(processorError)) {
+      if (
+        (processorError.response?.data as ResponseError).code === "OPENBRIDGE-1"
```

Comment:
I agree. I would try to avoid doing it manually if it's possible. I [opened a ticket](https://issues.redhat.com/browse/MGDOBR-669) a while back to set up scripts to automatically generate the error list like others SDKs do.

We could take care of it before doing [error handling consolidation](https://issues.redhat.com/browse/MGDOBR-715).

Reply:
Probably it is worth to add a _TODO_ (with the ticket you're referring) for catching it faster in the future

Reply:
Of course. Done.